client/debug/main.go

```diff
@@ -0,0 +1,178 @@
+package debug
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/spf13/cobra"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+)
+
+func Cmd(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "debug",
+		Short: "tool for helping with debugging your application",
+		RunE:  client.ValidateCmd,
+	}
+
+	cmd.AddCommand(PubkeyCmd(cdc))
+	cmd.AddCommand(AddrCmd())
+	cmd.AddCommand(RawBytesCmd())
+
+	return cmd
+}
+
+func PubkeyCmd(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "pubkey",
+		Short: "Decode a pubkey from hex, base64, or bech32",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			pubkeyString := args[0]
+			var pubKeyI crypto.PubKey
+
+			// try hex, then base64, then bech32
+			pubkeyBytes, err := hex.DecodeString(pubkeyString)
+			if err != nil {
+				var err2 error
+				pubkeyBytes, err2 = base64.StdEncoding.DecodeString(pubkeyString)
+				if err2 != nil {
+					var err3 error
+					pubKeyI, err3 = sdk.GetAccPubKeyBech32(pubkeyString)
+					if err3 != nil {
+						var err4 error
+						pubKeyI, err4 = sdk.GetValPubKeyBech32(pubkeyString)
+
+						if err4 != nil {
+							var err5 error
+							pubKeyI, err5 = sdk.GetConsPubKeyBech32(pubkeyString)
+							if err5 != nil {
+								return fmt.Errorf(`expected hex, base64, or bech32. Got errors:
+									hex: %v,
+									base64: %v
+									bech32 Acc: %v
+									bech32 Val: %v
+									bech32 Cons: %v`,
+									err, err2, err3, err4, err5)
+							}
+
+						}
+					}
+
+				}
+			}
+
+			var pubKey ed25519.PubKeyEd25519
+			if pubKeyI == nil {
+				copy(pubKey[:], pubkeyBytes)
+			} else {
+				pubKey = pubKeyI.(ed25519.PubKeyEd25519)
+				pubkeyBytes = pubKey[:]
+			}
+
+			pubKeyJSONBytes, err := cdc.MarshalJSON(pubKey)
+			if err != nil {
+				return err
+			}
+			accPub, err := sdk.Bech32ifyAccPub(pubKey)
+			if err != nil {
+				return err
+			}
+			valPub, err := sdk.Bech32ifyValPub(pubKey)
+			if err != nil {
+				return err
+			}
+
+			consenusPub, err := sdk.Bech32ifyConsPub(pubKey)
+			if err != nil {
+				return err
+			}
+			fmt.Println("Address:", pubKey.Address())
```

Comment:
Can you use `cmd.Println()` instead please? This would make it easier to test the cobra command's output like we do for `client/keys` cobra commands (hint: look at the `_test.go` files in the `client/keys` directory to see what I mean